Update default colors to match the Tomorrow Theme

bruvellu opened the pull request:
Dear Jaap,

As discussed in #1727, this PR proposes to update the default Zim color scheme to match the [Tomorrow Theme](https://github.com/chriskempson/tomorrow-theme). This palette provides a more balanced set of colors that adapt equally well for light and dark themes. It is also similar to the current theme, so it won't be a drastic change of style. I put some before/after screenshots below.

I would love to hear what you and the community think about these changes. There are some tweaks and ideas that I would like to discuss if they make sense to be included as defaults. But I'll add these updates later on. For now, any feedback on this initial version is welcome!

## Headings
![tomorrow-zim-headings](https://github.com/user-attachments/assets/a5e214c5-4cbd-4b86-b0e3-b9aab0835fb1)

## Formatting
![tomorrow-zim-formatting](https://github.com/user-attachments/assets/572cea13-214b-46b1-af93-e9c2852ef649)

## Task priority
![tomorrow-zim-priority](https://github.com/user-attachments/assets/4b658f6c-362a-4e77-a515-9b9ca1992ee7)

## Dark theme (formatting)
![tomorrow-zim-formatting-dark](https://github.com/user-attachments/assets/981d8963-aa09-4674-98a8-1a7e82f10418)

Cheers,
Buno

bruvellu committed:
Update default colors to match the Tomorrow Theme

UnisTorvalds commented:
I find that on light theme the contrast (therefore readability) is reduced, esp. with headlines, highlights, links, and task priorities.  I would see this as a regression.
However it's true that on dark theme readability is drastically improved.
I don't use dark theme so personally I am ambivalent to this change.  Would there be an easy (in app) way to change themes without swapping css files?

bruvellu committed:
Improve `pre` contrast with paragraph-background

The foreground color in multi-line verbatim block was too faint. I
changed it to the default text color and added a light gray background
to highlight the block using the paragraph-background property. This is
based on #512 and #983.

bruvellu committed:
Apply gray background to in-line verbatim

Same principle as the previous commit to improve the contrast for
verbatim text.

bruvellu committed:
Improve contrast of highlighted text

I've also added FIXME comments after noticing that `darktheme`
properties are not being properly applied when switching to dark mode on
my desktop. This needs some investigation.

bruvellu committed:
Add matching colors for different task checkboxes

This is more of an experiment with color-coding tasks. The colors match
the checkbox states fairly well (except the green). It looks better than
I imagined, but it might be too colorful for a default setting.

bruvellu commented:
Hi @UnisTorvalds, thanks for the input. Tomorrow is indeed a medium-contrast theme, so the colors are not so bright. However, I still find it comfortably readable and more balanced for the eyes compared to the default colors.

But I agree with you that some colors are a bit muted. Here's Tomorrow's blue `#4271ae` versus GitHub's links `#0969da`. We can tweak these specific colors to improve their contrast without altering the color scheme too much to make sure that Zim's default theme has good overall readability.

Alternatively, we could use another base color scheme. There are other options that might fit in case Tomorrow is not appealing for most (but I haven't tested them).

In any case, for now, the only way to change these colors in Zim is by editing the `style.conf` or CSS files (I believe), so not _in app_ options.

bruvellu commented:
I've made a few changes to the light theme to improve the contrast for highlighted and verbatim text. Colors for headings, tags, and links remain the same for now. But I did an experiment with color-coding the checked tasks. Please let me know what you think.

@jaap-karssenberg ~~I just noticed that `background[darktheme]` properties are not being applied. Could you take a look?~~ **Edit:** Sorry, they are applied when I manually set the `Prefer dark theme` option in Zim. I was testing by switching my desktop theme from light to dark (in this case, they are not applied).

![tomorrow-zim-formatting-update1](https://github.com/user-attachments/assets/30f45d3e-2499-4655-af2e-cfe22672a2a0)

bruvellu committed:
Improve highlight/priority contrast in darktheme

The colors red/orange/yellow of the Tomorrow theme are meant for
foreground text. When used as a background highlight with white text the
contrast was very low. I made darker shades of the original colors.

bruvellu committed:
Improve highlight/priority contrast for light theme

Same as the previous commit, but for the light theme. I made the
red/orange/yellow shades lighter when used as a background highlight
with a black foreground text.

bruvellu committed:
Make darktheme code/pre highlight brighter gray

The darker gray used as a background highlight for code and pre tags had
a strange blue tone. I desaturated it and made it slightly brighter.

bruvellu commented:
A few more updates. Tomorrow's colors are meant for foreground text. When I used them as background for highlighted text and priority tasks, they were either too dark (in the light theme) or too light (in the dark theme). Now, I tweaked them accordingly for better contrast, aiming for a [minimum contrast ratio](https://www.w3.org/TR/WCAG21/#contrast-minimum) around 5:1. Here's how the new version compares to Zim 0.75.2:

Task priorities light theme:
![tomorrow-zim-formatting-priority1](https://github.com/user-attachments/assets/50374688-2ab0-4dfe-ad3c-f4cff240269e)

Task priorities dark theme:
![tomorrow-zim-formatting-priority-dark1](https://github.com/user-attachments/assets/855546db-52a5-4b8e-852f-2f12c6df3c4a)

Formatting dark theme:
![tomorrow-zim-formatting-dark-formatting1](https://github.com/user-attachments/assets/4cde1a8c-3673-4c4a-91c1-172cec7e3f96)

bruvellu committed:
Desaturate completely pre/code foreground

bruvellu committed:
Desaturate strike foreground

bruvellu committed:
Improve contrast for strike text in light theme

bruvellu committed:
Improve orange contrast for tag and migrated task

bruvellu committed:
Annotate lighter versions for background highlight

bruvellu committed:
Use strike gray (better contrast) for inactive tasks

bruvellu committed:
Remove unused heading tag h6

bruvellu committed:
Remove outdated comments

bruvellu committed:
Standardize code styling

bruvellu commented:
A few more tweaks to the light theme, mostly to improve contrast for colored text. Here's how it currently looks:

![tomorrow-zim-formatting-update2](https://github.com/user-attachments/assets/7ede4669-cd69-478e-b15f-f732974c3dc9)

jaap-karssenberg commented:
I've also made a change to the default `style.conf`, it now supports different color tags for light and dark theme, removing the need for one set that works with both. I'm open to including more styles, but the readability for both light and dark themes should be addressed now.

bruvellu commented:
Hi @jaap-karssenberg, I don't understand what you mean. The modifications in this PR already use the new light/dark tags in `style.conf` and I've tweaked the colors of both variants to be consistent and have a good contrast for general use. This is how light/dark versions currently look:

![zim-lightdark](https://github.com/user-attachments/assets/f6545882-42e3-4847-8db4-742057318c10)

Let know if you want anything changed, or if you prefer not to have these as default, how should I submit as additional style.

Cheers!
Bruno

jaap-karssenberg commented:
Hi @bruvellu ,

Apologies, I was not reading carefully and took this as a "work around" for #1727 with one color set for both light and dark themes. Reading more closely I understand you work on two variants.

Previous I took the "Tango" palette as the basis, as that was positioned from the freedesktop.org side to be a good x-platform default palette. I think indeed many of your tweaks make sense and indeed improve the aesthetics.

Looking at the screenshots my personal preference would be to keep the brighter green from the original Tango palette for the headings - but this is also me being used to that color for the past 19 years, so might not be unbiased ;)  Also I would not prefer the strike-through & text color for tasks because it makes my notes look really busy.

Thinking what would be a good way to decide on aesthetics in an "objective" way.

Regards,

Jaap

jaap-karssenberg commented:
Playing around with the theme, but afraid I'm not convinced to change the default. I think the next step is to make a preference for the color scheme that can be changed via a dropdown or similar.

Kind regards,

Jaap

bruvellu commented:
Hi @jaap-karssenberg!

Yes, it makes sense to stick with the standard Tango colors as the default. The colored strikethroughs were an experimental idea I had while tweaking this theme over the past months. I ended up liking them as visual cues, but it can indeed feel busy with extensive task lists. I'll then close this PR and maintain the Tomorrow theme `style.conf` on a [separate repository](https://github.com/bruvellu/zim-tomorrow-theme). Do you have an idea already on the best way to implement the colorscheme preference? (perhaps something similar to the source view plugin?) I'd be happy to help to set this up.

Cheers,
Bruno

jaap-karssenberg commented:
Created #2766 to track adding additional style config to zim
